sdc-plugin/buffers.cc

```diff
@@ -15,10 +15,75 @@
  *  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
  *  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
+#include <cassert>
+#include <cmath>
 #include "buffers.h"
 
 const std::vector<std::string> Pll::inputs = {"CLKIN1", "CLKIN2"};
 const std::vector<std::string> Pll::outputs = {"CLKOUT0", "CLKOUT1", "CLKOUT2",
                                                "CLKOUT3", "CLKOUT4", "CLKOUT5"};
 const float Pll::delay = 0;
 const std::string Pll::name = "PLLE2_ADV";
+
+Pll::Pll(RTLIL::Cell* cell, float input_clock_period, float input_clock_shift) {
+    assert(RTLIL::unescape_id(cell->type) == "PLLE2_ADV");
+
+    FetchParams(cell);
+    if (clkin1_period != input_clock_period) {
```

Comment:
This should be a approx equal check, e.g. `fabs(a - b) < max(a, b)*10*epsilon`.  This prevents non-relevant rounding errors from triggering the error.

Reply:
Done. However I am not sure what is meant by the `10` in  `10*epsilon`.
I implemented it as `fabs(a-b) < max(a, b) * kApproxEqualFactor` where the `kApproxEqualFactor` is expressed in float, i.e. 0.01 means +/- 1%

Reply:
https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon

Reply:
Thanks, fixed.